### PR TITLE
Copy edit pass

### PR DESCRIPTION
**Pull request by gitKrystan**
1. Added chapter intro summaries and section headers
2. Did a general editorial pass, including rewording things for clarity as well as fixing typos

### CODE REVIEW

File: src/chapters/06-interactive-components.md [hunk 1]
```diff
@@ -43,6 +54,8 @@ Ember will create an *[instance][TODO: link to instance]* of the class whenever
 
 Here, in the *[component's constructor][TODO: link to component's constructor]*, we *[initialized][TODO: link to initialized]* the *[instance variable][TODO: link to instance variable]* `this.isLarge` with the value `false`, since this is the default state that we want for our component.
 
+## Accessing Instance Variable States from Templates
```
Comment (chancancode) [inline src/chapters/06-interactive-components.md]: "Instance Variable States" is redundant, either "Instance Variables" or "Instance States" are fine (probably the latter).

File: src/chapters/06-interactive-components.md [hunk 2]
```diff
@@ -2,13 +2,24 @@
...
 
 Sometimes, you want to associate some *[behavior][TODO: link to behavior]* with your components so that they can do more interesting things. For example, `<LinkTo>` can respond to clicks by changing the URL and navigating us to a different page.
 
 Here, we are going to do just that! We are going to implement the "View Larger" and "View Smaller" functionality, which will allow our users to click on a property's image to view a larger version, and click on it again to return to the smaller version.
 
-In other words, we want a way to _toggle_ the image between one of the two *[states][TODO: link to states]*. In order to do that, we need a way for the component to store two possible states, and to be aware which states it is currently in.
+In other words, we want a way to _toggle_ the image between one of the two *[states][TODO: link to states]*. In order to do that, we need a way for the component to store two possible states, and to be aware of which states it is currently in.
```
Comment (chancancode) [inline src/chapters/06-interactive-components.md]: I think this actually want to say "which state it is currently in" (singular "state")

File: src/chapters/06-interactive-components.md [hunk 3]
```diff
@@ -63,7 +76,7 @@ Let's update our template to use this state we just added:
 +{{/if}}
 ```
 
-In the template, we have access to the component's instance variables. The `{{#if ...}}...{{else}}...{{/if}}` *[conditionals][TODO: link to conditionals]* syntax allows us to render different content based on a condition (in this case, the value of the instance variable `this.isLarge`). Combining these two features, we can render either the small or the large version of the image accordingly.
+In the template, we have access to the component's instance variables. The `{{#if ...}}...{{else}}...{{/if}}` *[conditionals][TODO: link to conditionals]* syntax allows us to render different content based on a condition (in this case, the value of the instance variable `this.isLarge`). Note that we access instance variables in the template with `this.*`, not the `@argument` syntax you've seen for passed in arguments. Combining these two features, we can render either the small or the large version of the image accordingly.
```
Comment (chancancode) [inline src/chapters/06-interactive-components.md]:
```suggestion
In the template, we have access to the component's instance variables, using the same `this.isLarge` syntax. The `{{#if ...}}...{{else}}...{{/if}}` *[conditionals][TODO: link to conditionals]* syntax allows us to render different content based on a condition. In this case, the condition happens to be the `this.isLarge` instance variable. Combining these two features, we can render either the small or the large version of the image accordingly.
```

I am not sure that it is a good idea to call out "as opposed to the `@argument` syntax". It is actually a bit confusing for me to think of them as related at all. I understand where you are coming from, since in old-Ember everything is smooshed together, but I don't know if this is a thought that would come up for a new user reading this tutorial.

File: src/chapters/08-working-with-data.md [hunk 4]
```diff
@@ -194,9 +203,9 @@ wait  #qunit-banner.qunit-pass
 
 Now that we have things in place, let's do the fun part of removing *all* our hard-coded values from the model hook and actually fetch some data from the server!
 
-In a production app, the data that we'd fetch would most likely come from a remote API server. To avoid setting up an API server just for this tutorial, we will put some JSON data into the `public` folder instead. That way, we can still request these JSON data with regular HTTP requests &mdash; just like we would with a real API server  &mdash; but without having to write any server logic.
+In a production app, the data that we'd fetch would most likely come from a remote API server. To avoid setting up an API server just for this tutorial, we will put some JSON data into the `public` folder instead. That way, we can still request this JSON data with regular HTTP requests &mdash; just like we would with a real API server  &mdash; but without having to write any server logic.
```
Comment (chancancode) [inline src/chapters/08-working-with-data.md]: There are multiple JSON files, is this change correct?

File: src/chapters/08-working-with-data.md [hunk 5]
```diff
@@ -194,9 +203,9 @@ wait  #qunit-banner.qunit-pass
...
 
-But where will the data come from? You can <a href="/downloads/data.zip" download="data.zip">download this data file</a>, where we have prepared some JSON data and bundled them into a `.zip` file format. Extract its content into the `public` folder.
+But where will the data come from? You can <a href="/downloads/data.zip" download="data.zip">download this data file</a>, where we have prepared some JSON data and bundled it into a `.zip` file format. Extract its content into the `public` folder.
```
Comment (chancancode) [inline src/chapters/08-working-with-data.md]: Same

**Comment by chancancode** [comment]
Thanks! This looks great to me other than the few things I mentioned.

For the introduction paragraphs, I would like to mention what we are actually building. Something like this:

```
In this chapter, will we work on FEATURE XYZ, which allows the users to XYZ. This is what the app looks like when we are done:

<!-- TODO: add screen shot of the end state -->

While working on this, we will learn the following Ember concepts:

...
```

File: src/chapters/08-working-with-data.md (hunk 4 above)
Comment (gitKrystan) [reply src/chapters/08-working-with-data.md]: Depends on whom you ask. [OED](https://www.lexico.com/en/definition/data) says "data" is a mass noun and thus gets singular "this".

[APA](https://blog.apastyle.org/apastyle/2012/07/data-is-or-data-are.html), however, says to use "these data".

Dianne and I both made this face when we read "these data": 🤔

"This data" feels less formal too.

File: src/chapters/08-working-with-data.md (hunk 5 above)
Comment (gitKrystan) [reply src/chapters/08-working-with-data.md]: We'll duke it out above. :-P

File: src/chapters/08-working-with-data.md (hunk 4 above)
Comment (wycats) [reply src/chapters/08-working-with-data.md]: The AP styleguide says it's singular in non-academic contexts. https://twitter.com/apstylebook/status/1111694537020370945?lang=en

That sounds about right to me.

**Comment by gitKrystan** [comment]
@chancancode In chapter 8, should we provide more explanation about how the public folder works? We don't really explain _why_ we are able to fetch stuff at `/api` just because there is an `api` folder in the `public` folder. It feels a little magic.

**Comment by chancancode** [comment]
Did we not covet that in the orientation chapter already?

File: src/chapters/00-part-1.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 Welcome to the Ember Tutorial!
 
-In this tutorial, we will use Ember to build a professional looking application called Super Rentals. It's a website for browsing interesting places to stay at for your next vacation. Check out the [finished app](https://super-rentals-tutorial--ember-super-rentals.netlify.com) to get a sense of the scope of the project.
+In this tutorial, we will use Ember to build a professional looking application called Super Rentals. It's a website for browsing interesting places to stay for your next vacation. Check out the [finished app](https://super-rentals-tutorial--ember-super-rentals.netlify.com) to get a sense of the scope of the project.
```
Comment (vaidehijoshi) [inline src/chapters/00-part-1.md]:
```suggestion
In this tutorial, we will use Ember to build an application called Super Rentals. This will be a website for browsing interesting places to stay for your next vacation. Check out the [finished app](https://super-rentals-tutorial--ember-super-rentals.netlify.com) to get a sense of the scope of the project.
```

File: src/chapters/01-orientation.md [hunk 7]
```diff
@@ -1,4 +1,17 @@
-You can install the latest version of *[Ember CLI][TODO: link to Ember CLI]* by running the following command. If you've already done this by following the [Quick Start](../../getting-started/quick-start/) guide, feel free to skip ahead!
+In this chapter, you will install *[Ember CLI][TODO: link to Ember CLI]*, use it to generate a new Ember project, and add some basic templates and styles to your new app. By the end of this chapter, you should have a landing page with Professor Tomster's cute li'l face featured on it:
```
Comment (vaidehijoshi) [inline src/chapters/01-orientation.md]:
```suggestion
In this chapter, you will install *[Ember CLI][TODO: link to Ember CLI]*, use it to generate a new Ember project, and add some basic templates and styles to your new app. By the end of this chapter, you should have a landing page with Professor Tomster's cute little face featured on it:
```

File: src/chapters/01-orientation.md [hunk 8]
```diff
@@ -160,7 +175,9 @@ tree super-rentals -a -I "node_modules|.git|yarn.lock" --dirsfirst \
 tree super-rentals /F
 ```
 
-We will get to know the purposes of these files and folders as we go. For now, just know we will spend most of the time working within the `app` folder.
+We'll get to know the purposes of these files and folders as we go. For now, just know we'll spend most of the time working within the `app` folder.
```
Comment (vaidehijoshi) [inline src/chapters/01-orientation.md]:
```suggestion
We'll learn about the purposes of these files and folders as we go. For now, just know that we'll spend most of our time working within the `app` folder.
```

File: src/chapters/01-orientation.md [hunk 9]
```diff
@@ -185,10 +202,12 @@ visit http://localhost:4200/
 
 > Zoey says...
 >
-> The `localhost` address in URL means that you can only access the development server from your local machine. If you would like to share your work to the world, you will have to *[deploy][TODO: link to deploy]* your app to the public Internet. Don't worry, we will cover that in Part 2 of the tutorial.
+> The `localhost` address in the URL means that you can only access the development server from your local machine. If you would like to share your work with the world, you will have to *[deploy][TODO: link to deploy]* your app to the public Internet. Don't worry, we'll cover that in Part 2 of the tutorial.
```
Comment (vaidehijoshi) [inline src/chapters/01-orientation.md]:
```suggestion
> The `localhost` address in the URL means that you can only access the development server from your local machine. If you would like to share your work with the world, you will have to *[deploy][TODO: link to deploy]* your app to the public Internet. We'll cover how to do that in Part 2 of the tutorial.
```

File: src/chapters/01-orientation.md [hunk 10]
```diff
@@ -242,9 +263,9 @@ visit http://localhost:4200/
 git add app/templates/index.hbs
 ```
 
-Before we do anything else, let's add some styling to our app. We spend enough time staring at the computer screen as it is, we must protect our eyesight against unstyled markup!
+Before we do anything else, let's add some styling to our app. We spend enough time staring at the computer screen as it is, so we must protect our eyesight against unstyled markup!
 
-Fortunately, our designer sent us some CSS for us to use, so we can just go ahead <a href="/downloads/style.css" download="app.css">download the stylesheet file</a> and copy it into `app/styles/app.css`. This file has all the styles we need for building the rest of the app.
+Fortunately, our designer sent us some CSS for us to use, so we can just go ahead and <a href="/downloads/style.css" download="app.css">download the stylesheet file</a> and copy it into `app/styles/app.css`. This file has all the styles we need for building the rest of the app.
```
Comment (vaidehijoshi) [inline src/chapters/01-orientation.md]:
```suggestion
Fortunately, our designer sent us some CSS for us to use, so we can <a href="/downloads/style.css" download="app.css">download the stylesheet file</a> and copy it into `app/styles/app.css`. This file has all the styles we need for building the rest of the app.
```

File: src/chapters/01-orientation.md [hunk 11]
```diff
@@ -281,7 +302,7 @@ body {
 /* ...snip... */
 ```
 
-If you are familiar with CSS, feel free to customize them to your liking! Just keep in mind that you may see some visual differences going forward, if you choose to do so.
+If you are familiar with CSS, feel free to customize them to your liking! Just keep in mind that you may see some visual differences going forward, should you choose to do so.
```
Comment (vaidehijoshi) [inline src/chapters/01-orientation.md]:
```suggestion
If you are familiar with CSS, feel free to customize these styles to your liking! Just keep in mind that you may see some visual differences going forward, should you choose to do so.
```

File: src/chapters/07-reusable-components.md [hunk 12]
```diff
@@ -208,12 +229,14 @@ We just added a lot of behavior into a single component, so let's write some tes
...
 +    />`);
 +
-+    assert.dom('.map img').hasAttribute('src', /^https:\/\/api\.mapbox\.com/);
++    assert.dom('.map img').hasAttribute('src', /^https:\/\/api\.mapbox\.com/, 'the src starts with "https://api.mapbox.com"');
 +    assert.dom('.map img').hasAttribute('width', '150');
 +    assert.dom('.map img').hasAttribute('height', '120');
    });
 ```
 
+Note that the `hasAttribute` test helper from [qunit-dom][TODO: link to qunit-dom] supports using *[regular expressions][TODO: link to regular expressions]*. We used this feature to confirm that the `src` attribute starts with `https://api.mapbox.com/`, as opposed to requiring it to be an exact match against a string. This allows us to be reasonably confident that the code is working correctly but without getting overly detailed in the tests.
```
Comment (vaidehijoshi) [inline src/chapters/07-reusable-components.md]:
```suggestion
Note that the `hasAttribute` test helper from [qunit-dom][TODO: link to qunit-dom] supports using *[regular expressions][TODO: link to regular expressions]*. We used this feature to confirm that the `src` attribute starts with `https://api.mapbox.com/`, as opposed to requiring it to be an exact match against a string. This allows us to be reasonably confident that the code is working correctly, without being overly-detailed in our tests. 
```

File: src/chapters/07-reusable-components.md [hunk 13]
```diff
@@ -328,9 +353,9 @@ Note that we did not mark our getter as `@tracked`. Unlike instance variables, g
...
 
-Ember does this by automatically tracking any variables that were accessed while computing a getter's value. As long as the dependencies themselves are marked as `@tracked`, Ember will know exactly when to invalidate and re-render any templates that may potentially contain any "stale" (outdated) getter values. This feature is also know as *[auto-track][TODO: link to auto-track]*.
+Ember does this by automatically tracking any variables that were accessed while computing a getter's value. As long as the dependencies themselves are marked as `@tracked`, Ember will know exactly when to invalidate and re-render any templates that may potentially contain any "stale" (outdated) getter values. This feature is also know as *[auto-track][TODO: link to auto-track]*. Since all arguments (`this.args.*`) are implicitly marked as `@tracked` by the Glimmer component superclass that we inherited from, everything Just Works&trade;.
```
Comment (vaidehijoshi) [inline src/chapters/07-reusable-components.md]:
```suggestion
Ember does this by automatically tracking any variables that were accessed while computing a getter's value. As long as the dependencies themselves are marked as `@tracked`, Ember knows exactly when to invalidate and re-render any templates that may potentially contain any "stale" and outdated getter values. This feature is also known as *[auto-track][TODO: link to auto-track]*. All arguments that can be accessed from `this.args` (in other words, `this.args.*`) are implicitly marked as `@tracked` by the Glimmer component superclass. Since we inherited from that superclass, everything Just Works&trade;.
```

File: src/chapters/08-working-with-data.md [hunk 14]
```diff
@@ -312,10 +337,12 @@ We can handle it all in our model hook:
    }
 ```
 
-After parsing the JSON data, we extracted the nested `attributes` object, added back the missing `type` attribute manually, and then return it from the model hook. That way, the rest of our app won't have any idea that these difference ever existed.
+After parsing the JSON data, we extracted the nested `attributes` object, added back the missing `type` attribute manually, and then returned it from the model hook. That way, the rest of our app won't have any idea that this difference ever existed.
```
Comment (vaidehijoshi) [inline src/chapters/08-working-with-data.md]:
```suggestion
After parsing the JSON data, we extracted the nested `attributes` object, added back the missing `type` attribute manually, and then returned it from the model hook. That way, the rest of our app will have no idea that this difference ever existed.
```

File: src/chapters/07-reusable-components.md (hunk 13 above)
Comment (vaidehijoshi) [reply src/chapters/07-reusable-components.md]: Can feel free to tweak this. This edit is more to clarify what `this.args.*` is, since I'm not sure that beginners would necessarily know that it refers to "all arguments" if they hadn't seen that syntax before.

**Comment by vaidehijoshi** [comment]
thanks for these edits, @gitKrystan + Dianne! most of my notes are minor, trying to keep tense and tone consistent with the rest of the tutorial

File: src/chapters/07-reusable-components.md (hunk 13 above)
Comment (gitKrystan) [reply src/chapters/07-reusable-components.md]: Seems good. Makes it easier to `ctrl+f` too.

File: src/chapters/02-building-pages.md [hunk 15]
```diff
@@ -2,6 +2,19 @@
 ember server
 ```
 
+In this chapter, you will build the first few pages of your Ember app and set up links between them. By the end of this chapter, you should have new `/about` and `/contact` pages linked to your landing page:
```
Comment (chancancode) [inline src/chapters/02-building-pages.md]:
```suggestion
In this chapter, you will build the first few pages of your Ember app and set up links between them. By the end of this chapter, you should two new pages built – an "About" page and a "Contact" page. These pages will be linked to from your landing page:
```

The contact page's URL is actually _not_ `/contact`.

File: src/chapters/03-automated-testing.md [hunk 16]
```diff
@@ -2,6 +2,16 @@
 ember server
 ```
 
+In this chapter, you will use Ember's built-in testing framework to test your app, and in the process, you will learn about:
```
Comment (chancancode) [inline src/chapters/03-automated-testing.md]:
```suggestion
In this chapter, you will use Ember's built-in testing framework to write some automated tests for your app, and in the process, you will learn about:
```

File: src/chapters/05-more-about-components.md [hunk 17]
```diff
@@ -4,6 +4,16 @@ ember server
 
 It's time to finally work on the rentals listing!
 
+<!-- TODO: add screen shot of the end state -->
+
+While listing your rentals, you will learn about:
```
Comment (chancancode) [inline src/chapters/05-more-about-components.md]:
```suggestion
While building this list of rental properties, you will learn about:
```

File: src/chapters/08-working-with-data.md [hunk 18]
```diff
@@ -2,9 +2,24 @@
 ember server
 ```
 
+In this chapter, we will remove the hard-coded data from our `<Rental>` component. By the end, your app should look the same, but with the ability to `fetch` this data from an API.
```
Comment (chancancode) [inline src/chapters/08-working-with-data.md]:
```suggestion
In this chapter, we will remove the hard-coded data from our `<Rental>` component. By the end, your app would finally be displaying real data that came from the server!
```

It doesn't actually look the same, because the single duplicated listing will now be replaced with three "real" rental listing.

Comment (vaidehijoshi) [reply src/chapters/08-working-with-data.md]:
```suggestion
In this chapter, we will remove the hard-coded data from our `<Rental>` component. By the end, your app will finally display real data that will come from the server!
```

Comment (vaidehijoshi) [reply src/chapters/08-working-with-data.md]: "In this chapter, we will remove the hard-coded data from our `<Rental>` component. By the end, your app will finally display real data that comes from the server!"

**Comment by chancancode** [comment]
Thank you!
